### PR TITLE
use pipenv in image building dockerfile

### PR DESCRIPTION
**Pull request by Jimexist**
ready for review now

### CODE REVIEW

**Comment by jlewi** [comment]
Is this ready for review? I ask because of the TODO? If not please put wip in the title.

File: images/Dockerfile [hunk 1]
```diff
@@ -92,13 +92,16 @@ RUN cd /tmp && \
     mv ks_0.8.0_linux_amd64/ks /usr/local/bin && \
     chmod a+x /usr/local/bin/ks
 
-COPY ./requirements.txt /tmp/requirements.txt
+COPY ./Pipfile ./Pipfile.lock /tmp/
```
Comment (jlewi) [inline images/Dockerfile]: Does this get used anywhere?

**Comment by Jimexist** [comment]
@jlewi can you try to toggle and enable https://travis-ci.org/kubeflow/testing?

File: images/Dockerfile (hunk 1 above)
Comment (Jimexist) [reply images/Dockerfile]: it is used in https://github.com/kubeflow/testing/pull/36/files#diff-475e714b0c3678141c59dab9961e8841R103

**Comment by jlewi** [comment]
@Jimexist Sorry missed your first ping. Travis should be enabled now.

**Comment by jlewi** [comment]
@Jimexist Can you create instructions for updating/generating the Pipfile.

**Comment by Jimexist** [comment]
@jlewi will do. by the way any way to update the cla flag?

**Comment by Jimexist** [comment]
nevermind - i squashed locally

**Comment by Jimexist** [comment]
@jlewi seems CLA bot didn't re-check. Can you help manually merge?

**Comment by jlewi** [comment]
@Jimexist Can you update the Makefile so that you can build/push an image without tagging it latest? And then add a separate build command to tag it latest?

Then can you build/push an image to kubeflow-ci.
Then i can quickly check that the dependencies are working.

**Comment by jlewi** [comment]
@Jimexist Were you able to test this image? If you wanted to you could create an E2E test (an Argo workflow) that would build the image and then do a smoke test on that image just by running a simple python script that would try to import various dependencies.

**Comment by jlewi** [comment]
Any update?

**Comment by jlewi** [comment]
Thanks for updating it. Can you update the Makefile so I can try building an image and verify it works?

**Comment by jlewi** [comment]
@Jimexist the Makefile was most likely updated in another file. 
Can you resolve the conflicts?

**Comment by Jimexist** [comment]
@jlewi updated, could you take a look?

File: py/kubeflow/testing/util.py [hunk 2]
```diff
@@ -47,7 +46,7 @@ def run(command, cwd=None, env=None, polling_interval=datetime.timedelta(seconds
       lines.append("{0}={1}".format(k, env[k]))
     logging.info("Running: Environment:\n%s", "\n".join(lines))
 
-  log_file = None
+  # log_file = None
```
Comment (jlewi) [inline py/kubeflow/testing/util.py]: Can you just delete this line?

**Comment by jlewi** [comment]
Can you do 
make push
to build a new image (but not tag it latest) and then

1. Can you check that the python packages can be imported
2. Post a link to the image here?

IIRC in a previous attempt there was a problem with pipenv not actually installing the packages so it would be good to verify its working now.

**Comment by Jimexist** [comment]
```
docker history gcr.io/kubeflow-ci/test-worker:v20180328-01bb4bc-e3b0c4
IMAGE               CREATED              CREATED BY                                      SIZE                COMMENT
3a7c875069b3        48 seconds ago       /bin/sh -c #(nop)  ENTRYPOINT ["/usr/local/b…   0B
3126fcc433b1        48 seconds ago       /bin/sh -c #(nop)  ENV PYTHONPATH=/src/kubef…   0B
a438cc458dd3        48 seconds ago       /bin/sh -c #(nop)  ENV USER=root                0B
ec4115b77de0        48 seconds ago       /bin/sh -c curl  https://get.docker.com/ | sh   185MB
52def2fd6e26        About a minute ago   /bin/sh -c chmod a+x /usr/local/bin/run_work…   520B
0de0b276dbea        About a minute ago   /bin/sh -c #(nop) COPY file:7335db716fdba146…   520B
3dcd1edd68e8        About a minute ago   /bin/sh -c chmod a+x /usr/local/bin/checkout…   2.76kB
6df67c4e7ee4        About a minute ago   /bin/sh -c #(nop) COPY file:d51a70df6dc36afb…   2.76kB
9ba7ac0ed83f        About a minute ago   /bin/sh -c cd /tmp/ &&     pip install -U pi…   244MB
bdc9c5752ed9        2 minutes ago        /bin/sh -c #(nop) COPY multi:848bf0c8dcf5bcf…   20.4kB
8332bcd7aafe        2 minutes ago        /bin/sh -c cd /tmp &&     wget https://githu…   10.3MB
d0c439ae8bbb        3 minutes ago        /bin/sh -c cd /tmp &&     wget -O ks.tar.gz …   49.3MB
9c87f1d5fd47        3 minutes ago        /bin/sh -c cd /tmp &&     wget -O glide-v0.1…   14.4MB
210187e5dc3c        4 minutes ago        /bin/sh -c curl -sS http://dl.yarnpkg.com/de…   6.18MB
60481f2a0a57        4 minutes ago        /bin/sh -c curl -sL https://deb.nodesource.c…   101MB
c5b92598233a        4 minutes ago        /bin/sh -c helm init --client-only              1.41MB
e9f542d9786b        4 minutes ago        /bin/sh -c wget -O /tmp/get_helm.sh     http…   64MB
977f59d7bbbb        4 minutes ago        /bin/sh -c wget -q https://dl.google.com/dl/…   366MB
169c06dcd53b        5 minutes ago        /bin/sh -c #(nop)  ENV PATH=/usr/local/go/bi…   0B
2d748b2b67fc        5 minutes ago        /bin/sh -c cd /tmp &&     wget -O /tmp/go.ta…   402MB
2320f38cc33f        5 minutes ago        /bin/sh -c set -ex     && apt-get update -yq…   378MB
b10bf3addc52        6 minutes ago        /bin/sh -c #(nop)  ENV LANG=C.UTF-8             0B
ad8d529b2fed        6 minutes ago        /bin/sh -c #(nop)  ENV LC_ALL=C.UTF-8           0B
361dbd1c5920        6 minutes ago        /bin/sh -c #(nop)  ENV TERM=linux               0B
2a9dbdd5af37        6 minutes ago        /bin/sh -c #(nop)  ENV DEBIAN_FRONTEND=nonin…   0B
1c15a190310f        6 minutes ago        /bin/sh -c #(nop)  MAINTAINER Jeremy Lewi       0B
f975c5035748        3 weeks ago          /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B
<missing>           3 weeks ago          /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B
<missing>           3 weeks ago          /bin/sh -c sed -i 's/^#\s*\(deb.*universe\)$…   2.76kB
<missing>           3 weeks ago          /bin/sh -c rm -rf /var/lib/apt/lists/*          0B
<missing>           3 weeks ago          /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B
<missing>           3 weeks ago          /bin/sh -c #(nop) ADD file:c753df38640ab6e24…   112MB
```

**Comment by Jimexist** [comment]
```
docker run --rm -it --entrypoint python gcr.io/kubeflow-ci/test-worker:v20180328-01bb4bc-e3b0c4 -V
Python 2.7.12
```

**Comment by Jimexist** [comment]
```
docker run --rm -it --entrypoint python3 gcr.io/kubeflow-ci/test-worker:v20180328-01bb4bc-e3b0c4 -V
Python 3.5.2
```

**Comment by Jimexist** [comment]
```docker run --rm -it --entrypoint python3 gcr.io/kubeflow-ci/test-worker:v20180328-01bb4bc-e3b0c4
Python 3.5.2 (default, Nov 23 2017, 16:37:01)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import kubernetes
>>> kubernetes.__version__
'4.0.0'
````

**Comment by Jimexist** [comment]
@jlewi please take a look thanks

**Comment by jlewi** [comment]
Fantastic thank you so much.
/lgtm
/approve

**Comment by jlewi** [comment]
Merging manually because of the CLA check.

**Pull request merged**